front-end-tools/src/components/DeployComponent.tsx

```diff
@@ -0,0 +1,268 @@
+    isTestnet: boolean;
+    setContracts: (arg0: string[]) => void;
+    setEmbeddedModuleSchemaBase64Init: (arg0: string) => void;
+    setModuleReferenceDeployed: (arg0: string | undefined) => void;
+    setModuleReferenceCalculated: (arg0: string) => void;
+    moduleReferenceDeployed: string | undefined;
+    moduleReferenceCalculated: string | undefined;
+}
+
+export default function DeployComponenet(props: ConnectionProps) {
+    const {
+        isTestnet,
+        client,
+        connection,
+        account,
+        setContracts,
+        setModuleReferenceDeployed,
+        moduleReferenceDeployed,
+        setModuleReferenceCalculated,
+        moduleReferenceCalculated,
+        setEmbeddedModuleSchemaBase64Init,
+    } = props;
+
+    type FormType = {
+        file: FileList | undefined;
+    };
+    const form = useForm<FormType>();
+
+    const [transactionErrorDeploy, setTransactionErrorDeploy] = useState<string | undefined>(undefined);
+    const [uploadError, setUploadError] = useState<string | undefined>(undefined);
+    const [isModuleReferenceAlreadyDeployedStep1, setIsModuleReferenceAlreadyDeployedStep1] = useState(false);
+    const [txHashDeploy, setTxHashDeploy] = useState<string | undefined>(undefined);
+    const [base64Module, setBase64Module] = useState<string | undefined>(undefined);
+    const [viewErrorModuleReference, setViewErrorModuleReference] = useState<string | undefined>(undefined);
+
+    // Refresh moduleReference periodically.
+    // eslint-disable-next-line consistent-return
+    useEffect(() => {
+        if (connection && client && account && txHashDeploy !== undefined) {
+            const interval = setInterval(() => {
+                console.log('refreshing_moduleReference');
+                client
+                    .getBlockItemStatus(txHashDeploy)
+                    .then((report) => {
+                        if (report !== undefined) {
+                            setViewErrorModuleReference(undefined);
+                            if (
+                                report.status === 'finalized' &&
+                                report.outcome.summary.type === TransactionSummaryType.AccountTransaction &&
+                                report.outcome.summary.transactionType === TransactionKindString.DeployModule
+                            ) {
+                                setModuleReferenceDeployed(report.outcome.summary.moduleDeployed.contents);
+                                clearInterval(interval);
+                            }
+                        }
+                    })
+                    .catch((e) => {
+                        setModuleReferenceDeployed(undefined);
+                        setViewErrorModuleReference((e as Error).message);
+                        clearInterval(interval);
+                    });
+            }, REFRESH_INTERVAL.asMilliseconds());
+        }
+    }, [connection, account, client, txHashDeploy]);
```

Comment:
Is it necessary to rerun this if `account` changes? This seems to only depend on the txHash, connection and client.

Reply:
removed